test/jdk/java/nio/channels/etc/ProtocolFamilies.java

```diff
@@ -39,10 +39,12 @@
 import static java.net.StandardProtocolFamily.INET;
 import static java.net.StandardProtocolFamily.INET6;
 import static jdk.test.lib.net.IPSupport.*;
+import static org.testng.Assert.assertNull;
 import static org.testng.Assert.assertThrows;
 
 /*
  * @test
+ * @bug 8254692
```

Comment:
test/jdk/java/nio/channels/spi/SelectorProvider/inheritedChannel would be a more appropriate location to test this.

I assume you modified etc/ProtocolFamilies.java you because it has a custom provider that doesn't override this method but this test has the summary "Test SocketChannel, ServerSocketChannel and DatagramChannel with various ProtocolFamily combinations" so not really the right place.

Reply:
The test method being changed here assets other aspects of the default implementation of `SelectorProvider` - this is just another case. I guess is depends on ones view, whether these checks should be co-located based on functionality or based on category of behaviour.  It might be best to create a new test/jdk/java/nio/channels/spi/SelectorProvider/DefaultImplementationTest.java to hold such, and move the other assertions there too.  But the higher-order bit is that we have tests.

Reply:
ProtocolFamilies/testCustomProvider exercises the methods that take a protocol family, it's not really the right place to exercise the inheritedChannel method. Can we add a test to test/jdk/java/nio/channels/spi/SelectorProvider/inheritedChannel instead?

Reply:
No disagreement about testCustomProvider not being the most appropriate place to test the default implementation of inheritedChannel. What I was suggesting was to move *all* the assertions related to the default implementation of SelectorProvider to a single new test located under test/jdk/java/nio/channels/spi/SelectorProvider/, rather than splitting the tests for the default implementation across several different tests and directory hierarchies.

Reply:
A new test/jdk/java/nio/channels/spi/SelectorProvider that covers all the assertions would be great, thanks!

Reply:
Thank you Alan and Chris for reviewing, I moved all the assertions related to the default implementation from ProtocolFamilies to a new test class as suggested and removed unused imports in ProtocolFamilies